doc/platform_api/CMIS_Diagnostic_Monitoring_Overview_in_SONiC.md

```diff
@@ -0,0 +1,2309 @@
+- `module_state_changed`
+- `txfault{lane_num}`
+- `rxlos{lane_num}`
+- `txlos_hostlane{lane_num}`
+- `txcdrlol_hostlane{lane_num}`
+- `tx_eq_fault{lane_num}`
+- `rxcdrlol{lane_num}`
+
+Example
+
+The following fields related to `datapath_firmware_fault` are updated in the `redis-db` during a link down event:
+
+- `datapath_firmware_fault` in `TRANSCEIVER_STATUS` table
+- `datapath_firmware_fault` in `TRANSCEIVER_STATUS_FLAG_CHANGE_COUNT` table
+- `datapath_firmware_fault` in `TRANSCEIVER_STATUS_FLAG_SET_TIME` table
+- `datapath_firmware_fault` in `TRANSCEIVER_STATUS_FLAG_CLEAR_TIME` table
+
+#### 4.2.3 Details of Flag Analysis of Tables
+
+**Note**: For simplicity, this section uses DOM as an example. However, the same analysis is applicable for VDM as well.
+
+**Purpose of Flag Analysis:**
+
+The purpose of flag analysis is to track the status of various parameters and to count the number of times each DOM flag has changed. It also records the timestamp when each DOM flag was set and cleared.
+
+**Tables Used for Flag Analysis:**
+
+- `TRANSCEIVER_DOM_FLAG`: This table stores flags indicating the status of various DOM parameters.
+- `TRANSCEIVER_DOM_FLAG_CHANGE_COUNT`: This table keeps a count of how many times each DOM flag has changed. Upon initialization, the count is set to 0.
+- `TRANSCEIVER_DOM_FLAG_SET_TIME`: This table records the timestamp (in local timezone) when each DOM flag was set. The timestamp is recorded in the format `Day Mon DD HH:MM:SS YYYY`. During initialization, the timestamp is set to `never` if the flag is not set.
+- `TRANSCEIVER_DOM_FLAG_CLEAR_TIME`: This table records the timestamp (in local timezone) when each DOM flag was cleared. The timestamp is recorded in the format `Day Mon DD HH:MM:SS YYYY`. During initialization, the timestamp is set to `never` if the flag is set.
+
+**Example of Table Updates:**
+
+- **TRANSCEIVER_DOM_FLAG_CHANGE_COUNT:**
+  - Each time a flag in the `TRANSCEIVER_DOM_FLAG` table changes (either set or cleared), the corresponding count in this table is incremented.
+- **TRANSCEIVER_DOM_FLAG_SET_TIME:**
+  - When a flag is set in the `TRANSCEIVER_DOM_FLAG` table, the current timestamp (based on local timezone) is recorded in this table.
+- **TRANSCEIVER_DOM_FLAG_CLEAR_TIME:**
+  - When a flag is cleared in the `TRANSCEIVER_DOM_FLAG` table, the current timestamp (based on local timezone) is recorded in this table.
+
+#### 4.2.4 Flag Change Count and Time Set/Clear Behavior During `xcvrd` Restart
+
+During `xcvrd` stop, the `TRANSCEIVER_DOM_FLAG_CHANGE_COUNT`, `TRANSCEIVER_DOM_FLAG_SET_TIME`, and `TRANSCEIVER_DOM_FLAG_CLEAR_TIME` tables are deleted by the `xcvrd` process. When `xcvrd` is restarted, the `TRANSCEIVER_DOM_FLAG_CHANGE_COUNT`, `TRANSCEIVER_DOM_FLAG_SET_TIME`, and `TRANSCEIVER_DOM_FLAG_CLEAR_TIME` tables are recreated and the flag change count and set/clear time are updated based on the current flag status (i.e. the value of these fields are not cached between `xcvrd` restarts).
```

Comment:
Do we want to differentiate between cold reboot and warm reboot scenarios?

Reply:
@keboliu with the first draft of this HLD, we do not plan to preserve the values of these tables between warm reboot. This will be in line to this section since warm reboot will cause xcvrd kill followed by xcvrd startup when the device reboots.
Preserving the values of these tables between warm reboot and warm restart of xcvrd (by warm restart of xcvrd, I mean that link flap is not expected on ports) can be taken as an enhancement in future.